### DISCUSSION TITLE
Mesh Central Install on ubuntu VM, running through NGNIX Proxy

### DISCUSSION THREAD

**Discussion by Acsriot94** [Ylianst/MeshCentral#6267] [General]
Hello everyone,

I've been running Meshcentral on Truecharts for a few years but with the recent changes I'm trying to migrate off of the TC install and into a VM. I have everything installed and playing nice but I've found that if my server has any issue whatsoever (Connectivity, power) the agents show offline - and the only remedy is manually reconnecting on the clients or when the clients restart they re-connect to Meshcentral.

My previous install of Mesh central would auto-reconnect to all the agents in a few minutes after re-establishing connect or coming back from a restart

Config

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "mesh.wilddragon.net",
"webRTC": true,
"AgentPing": 15,
"AgentPong": 15,
"BrowserPing": 15,
"BrowserPong": 15,
"sessionTime": 60,
  "agentCoreDump": false,
    "agentLogDump": false,
    "agentWsCompression": true,
  "allowHighQualityDesktop": true,
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
        "cookieipcheck": false,
        "_ignoreagenthashcheck": true,
        "mpsport": 44330,
        "mpsaliasport": 4433,
        "mpstlsoffload": true,
        "tlsoffload": "https://mesh.wilddragon.net",
    "port": 444,
    "aliasPort": 443,
    "redirPort": 800,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "Wild Dragon",
      "_title2": "Dragon Central",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
         "certurl": "https://mesh.wilddragon.net"
    }
},
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}

```

**Comment by si458** [comment]
Without knowing a little more about ur setup, vm specs, reverse proxy (I'm guessing based on config.json), it's hard to say

For example, you don't really need all the `ping/pong` options, as they just help keep things alive, but can also cause problems

**Comment by Acsriot94** [comment]
Truenas System which has a Local address of 10.0.25.5 -- NGNIX is running in the apps pool of Truenas on port 30021 & 30022 (HTTP & HTTPS). I have those ports forwarded to 443 & 80 externally. Meshcentral is running on uBunutu, followed the instructions for 18.04 on the Meshcentral documentation. 

I'll go ahead and remove the ping/pong and see if that helps.
